src/datumaro/components/media.py

```diff
@@ -330,6 +336,18 @@ def data(self) -> Optional[np.ndarray]:
             self._size = tuple(map(int, data.shape[:2]))
         return data
 
+    @property
+    def size(self) -> Optional[Tuple[int, int]]:
+        """Returns (H, W)"""
+
+        if self._size is None:
+            if _HAS_PIL:
+                w, h = PILImage.open(self.path).size
+                self._size = (h, w)
+            else:
+                _ = super().size
+        return self._size
```

Comment:
Hi @imyhxy,

Thanks for your good suggestion. However, in this PR https://github.com/openvinotoolkit/datumaro/pull/1246, we changed Datumaro image backend as selectable by the context variables. In addition, it would be good to extract a image size from the file header using the external library, e.g. https://pypi.org/project/imagesize,

```suggestion
import imagesize

...

    def size(self) -> Optional[Tuple[int, int]]:
        """Returns (H, W)"""

        if self._size is None:
            try:
                width, height = imagesize.get(self.path)
                self._size = (height, width)
            except Exception:
                _ = super().size
        return self._size
```

Would you please update your PR in this way? You might have to add that library to our `requirements.txt`.